File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/command/ModifierCommandExtension.java
package de.cubbossa.pathfinder.command;

import de.cubbossa.pathfinder.group.Modifier;
import dev.jorel.commandapi.arguments.Argument;
import dev.jorel.commandapi.executors.CommandExecutor;
import net.kyori.adventure.text.ComponentLike;

import java.util.function.Function;

public interface ModifierCommandExtension<M extends Modifier> {

  Argument<?> registerAddCommand(Argument<?> tree, Function<M, CommandExecutor> consumer);

  ComponentLike toComponents(M modifier);
}

File: pathfinder-core/src/main/java/de/cubbossa/pathfinder/command/ModifierCommandExtension.kt
package de.cubbossa.pathfinder.command

import de.cubbossa.pathfinder.group.Modifier
import dev.jorel.commandapi.arguments.Argument
import dev.jorel.commandapi.executors.CommandExecutor
import net.kyori.adventure.text.ComponentLike

interface ModifierCommandExtension<M : Modifier> {
    fun registerAddCommand(
        tree: Argument<*>,
        consumer: (M) -> CommandExecutor
    ): Argument<*>

    fun toComponents(modifier: M): ComponentLike
}
